src/main/java/qwerty/Storage.java

```diff
@@ -0,0 +1,135 @@
+package qwerty;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import qwerty.task.Deadline;
+import qwerty.task.Event;
+import qwerty.task.Task;
+import qwerty.task.Todo;
+
+public class Storage {
+    /** The file path of the save file */
+    private String savePath;
+
+    public Storage(String filePath) {
+        this.savePath = filePath;
+    }
+
+    /**
+     * Saves the current list of tasks in the given TaskList to the hard drive.
+     * Writes into a file at the specified path in savePath.
+     * If the file does not exist, create it.
+     *
+     * @param taskList The TaskList object.
+     */
+    public void saveTasks(TaskList taskList) {
+        File file = new File(savePath);
+
+        if (file.exists()) {
+            // Write the generated save string to the file
+            try {
+                FileWriter writer = new FileWriter(savePath);
+                writer.write(taskList.generateSaveString());
+                writer.close();
+            } catch (IOException e) {
+                System.out.println("Could not write to save file: " + e.getMessage());
+            }
+        } else {
+            // Create a new file and retry saving
+            try {
+                file.createNewFile();
+                saveTasks(taskList);
+            } catch (IOException e) {
+                System.out.println("Could not create save file: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Loads tasks into the given TaskList.
+     * Parses the string contents of the file located at the specified savepath.
+     * If the file does not exist, do nothing.
+     * If certain arguments are missing, prints an error statement.
+     *
+     * @param taskList The TaskList object to load into.
+     */
+    public void loadTasks(TaskList taskList) {
+        File file = new File(savePath);
+
+        if (file.exists()) {
+            try {
+                // Initialise scanner and line number counter
+                Scanner scanner = new Scanner(file);
+                int lineNumber = 0;
+
+                // Create a new task from parsing the details in each line and add it to the task list
+                while (scanner.hasNextLine()) {
+                    lineNumber++;
+                    try {
+                        // Parse the string in each line and match it to a task type
+                        String[] args = scanner.nextLine().split("\\|");
+                        switch (args[0]) {
```

Comment:
Perhaps you can consider extracting the logic into a private method, as right now it is a little confusing due to the deep indentation.